### PR TITLE
chore: orion --> server

### PR DESCRIPTION
**Pull request by jamiezieziula**
Rebrands `orion` references to `server` and other minor changes

### CODE REVIEW

File: .github/workflows/helm-release.yaml [hunk 1]
```diff
@@ -75,12 +75,12 @@ jobs:
           SIGN_KEYRING: ${{ env.SIGN_KEYRING }}
           SIGN_PASSPHRASE_FILE: ${{ env.SIGN_PASSPHRASE_FILE }}
 
-      - name: Package Orion helm chart
+      - name: Package server helm chart
```
Comment (jamiezieziula) [inline .github/workflows/helm-release.yaml:78]:
```suggestion
      - name: Package Server helm chart
```

File: .github/workflows/server-lint-and-test.yaml [hunk 2]
```diff
@@ -1,11 +1,11 @@
-name: Lint and Test Prefect Orion Chart
+name: Lint and Test Prefect server Chart
```
Comment (jamiezieziula) [inline .github/workflows/server-lint-and-test.yaml:1]:
```suggestion
name: Lint and Test Prefect Server Chart
```

File: charts/prefect-server/values.yaml [hunk 3]
```diff
@@ -11,31 +11,31 @@ commonLabels: {}
...
     ##   - myRegistryKeySecretName
-    # -- orion image pull secrets
+    # -- server image pull secrets
     pullSecrets: []
-    # -- enable orion image debug mode
+    # -- enable server image debug mode
     debug: false
 
-  # -- sets PREFECT_ORION_UI_API_URL; should be publicly accessible API URL; UI will not work unless set
+  # -- sets PREFECT_server_UI_API_URL; should be publicly accessible API URL; UI will not work unless set
```
Comment (jamiezieziula) [inline charts/prefect-server/values.yaml:35]:
```suggestion
  # -- sets PREFECT_SERVER_UI_API_URL; should be publicly accessible API URL; UI will not work unless set
```

**Review by parkedwards — APPROVED** [review]

**Comment by jamiezieziula** [comment]
will want signoff from @madkinsz

**Review by zanieb — APPROVED** [review]

**Comment by madkinsz** [comment]
@jamiezieziula this is live now :) is there a minimum prefect version we can/should enforce somewhere?

**Pull request merged**